src/gen.rs

```diff
@@ -47,8 +64,31 @@ impl Cmd {
             }
             Err(e) => {
                 println!("{}", code_raw);
-                Err(Error::FormatRustError(e))
+                Err(Error::FormatRust(e))
             }
         }
     }
+
+    pub fn generate_json(&self) -> Result<(), Error> {
+        let contents = fs::read(&self.wasm).map_err(Error::Io)?;
+        let h = Host::default();
+        let vm = Vm::new(&h, [0; 32].into(), &contents).map_err(Error::Host)?;
+
+        if let Some(spec) = vm.custom_section("contractspecv0") {
+            let mut cursor = Cursor::new(spec);
+            let json_entries: Result<Vec<json::Entry>, Error> =
+                ScSpecEntry::read_xdr_iter(&mut cursor)
+                    .map(|spec_entry| {
+                        json::Entry::try_from(&spec_entry.map_err(Error::Xdr)?).map_err(Error::Xdr)
+                    })
+                    .collect();
+            println!(
+                "{}",
+                serde_json::to_string(&json_entries?).map_err(Error::Json)?
+            );
```

Comment:
We should be able to call into the soroban-spec crate to do most of this work. I realize when we added the JSON logic we didn't add the same functions the rust version had, so I just added them in https://github.com/stellar/rs-soroban-sdk/pull/498, and will change this to use them and merge.